### PR TITLE
MULTIARCH-4185: Add ARM CP+ AMD DP e2e jobs

### PR DESCRIPTION
**Pull request by LiangquanLi930**
job link: https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/50523/rehearse-50523-periodic-ci-openshift-openshift-tests-private-release-4.16-arm64-nightly-azure-ipi-ovn-hypershift-mgmt-f7/1795034489940873216 
```
➜  export KUBECONFIG=kubeconfig          
➜  oc get clusterversion version                                     
NAME      VERSION                                    AVAILABLE   PROGRESSING   SINCE   STATUS
version   4.16.0-0.nightly-arm64-2024-05-27-050259   True        False         60m     Cluster version is 4.16.0-0.nightly-arm64-2024-05-27-050259
➜  oc get node                                                      
NAME                                                      STATUS   ROLES                  AGE   VERSION
ci-op-w7dqklqz-3b8d6-mqc95-master-0                       Ready    control-plane,master   88m   v1.29.5+87992f4
ci-op-w7dqklqz-3b8d6-mqc95-master-1                       Ready    control-plane,master   88m   v1.29.5+87992f4
ci-op-w7dqklqz-3b8d6-mqc95-master-2                       Ready    control-plane,master   88m   v1.29.5+87992f4
ci-op-w7dqklqz-3b8d6-mqc95-worker-southcentralus1-sfl7x   Ready    worker                 76m   v1.29.5+87992f4
ci-op-w7dqklqz-3b8d6-mqc95-worker-southcentralus2-6sjjd   Ready    worker                 76m   v1.29.5+87992f4
ci-op-w7dqklqz-3b8d6-mqc95-worker-southcentralus3-mwxdq   Ready    worker                 76m   v1.29.5+87992f4    
➜   oc get node -ojsonpath='{.items[*].status.nodeInfo.architecture}'
arm64 arm64 arm64 arm64 arm64 arm64        
➜ oc get hostedcluster -A        
NAMESPACE   NAME                   VERSION                                    KUBECONFIG                              PROGRESS    AVAILABLE   PROGRESSING   MESSAGE
clusters    3f0fd8a9f4c90ce40464   4.16.0-0.nightly-multi-2024-05-27-013444   3f0fd8a9f4c90ce40464-admin-kubeconfig   Completed   True        False         The hosted control plane is available
                                                                                                                             
➜  export KUBECONFIG=hostedcluster.kubeconfig
➜  oc get node
NAME                                  STATUS   ROLES    AGE   VERSION
3f0fd8a9f4c90ce40464-f412695a-g4pxm   Ready    worker   34m   v1.29.5+87992f4
3f0fd8a9f4c90ce40464-f412695a-hbtht   Ready    worker   34m   v1.29.5+87992f4
3f0fd8a9f4c90ce40464-f412695a-l5k92   Ready    worker   34m   v1.29.5+87992f4
➜  oc get node -ojsonpath='{.items[*].status.nodeInfo.architecture}'
amd64 amd64 amd64                                                                                                                                                                              
```

### CODE REVIEW

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-arm64-nightly-azure-ipi-ovn-hypershift-mgmt-f28

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-arm64-nightly-azure-ipi-ovn-hypershift-mgmt-f28

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-arm64-nightly-azure-ipi-ovn-hypershift-mgmt-f28

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-arm64-nightly-azure-ipi-ovn-hypershift-mgmt-f28

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-arm64-nightly-azure-ipi-ovn-hypershift-mgmt-f28

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-arm64-nightly-azure-ipi-ovn-hypershift-mgmt-f28

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-azure-ipi-ovn-hypershift-mgmt-f28

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-arm64-nightly-azure-ipi-ovn-hypershift-mgmt-f28

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-arm64-nightly-azure-ipi-ovn-hypershift-mgmt-f7

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-arm64-nightly-azure-ipi-ovn-hypershift-mgmt-f7

**Comment by LiangquanLi930** [comment]
@liangxia Could you help to review this PR? Thanks

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__arm64-nightly.yaml [hunk 1]
```diff
@@ -198,6 +204,56 @@ tests:
     test:
     - ref: openshift-extended-test-longduration
     workflow: cucushift-installer-rehearse-aws-ipi-private
+- as: azure-ipi-ovn-hypershift-guest-f7
```
Comment (liangxia) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__arm64-nightly.yaml:207]: According to the team agreement, if a test support both amd64 and arm64, then we move the test to multi. I'm just not sure if this rule applies to hypershift.

/cc @jianlinliu

Comment (YuLi517) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__arm64-nightly.yaml:207]: @liangxia the agreement is still available for HyperShift. What this PR doing is for Multi-Arch, ARM for management cluster and X86 for hosted cluster. The agreement will be available if HyperShift need to make hosted cluster as heterogeneous.

Comment (bryan-cox) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__arm64-nightly.yaml:207]: Just to add some clarification - only a multi arch release image is needed for a HostedCluster to have heterogenous NodePools. The arch of the mgmt cluster doesn't particularly matter.

**Pull request merged**